doc/feature/spikeReport.md

```diff
@@ -0,0 +1,450 @@
+         *
+         * @throws std::runtime_eror is the report was opened in write mode.
+         */
+        Spikes getSpikes( float startTime, float endTime,
+                          const uint32_t timeout = LB_TIMEOUT_INDEFINITE );
+
+        /*
+         * For reports open in write mode, write the given spikes to the output.
+         *
+         * @throws std::runtime_eror if the report was opened in read mode.
+         * @throws std::runtime_eror when writing to a stream if the spikes are
+         *         not sorted or any timestamp is <= to the highest timestamp
+         *         written by previous calls. This is required to enforce the
+         *         post-conditions of getSpikes in the receiving end.
+         */
+        void writeSpikes( const const std::vector< float, uint32_t >& spikes );
+
+        /* Overload of the method above provided for convenience */
+        void writeSpikes( const Spikes& spikes );
+
+        /*
+         * Return the start time of the report if known, UNDEFINED_TIMESTAMP
+         * otherwise.
+         *
+         * In file based reports it's the time of the first spike or undefined
+         * if the report is empty.
+         * In stream based reports it's always 0.
+         */
+        float getStartTime() const;
+
+        /*
+         * Return the end time of the report if known, UNDEFINED_TIMESTAMP
+         * otherwise.
+         *
+         * In file based reports the end time is the timestamp of the last spike
+         * plus an epsilon, or undefined if the report is empty or it cannot be
+         * be decided without loading the whole data.
+         * In stream based reports it's always undefined until the report is
+         * closed. When the report is closed it will be the latest timestamp
+         * reported by the sender or the time of the last spike + epsilon,
+         * whichever is larger; or 0 if no information was received.
+         *
+         * To guarantee const-correctness the value returned by this function
+         * can only change when getSpikes is called.
+         */
+        float getEndTime() const;
+
+        /*
+         * Closes the report.
+         *
+         * Only meaningful for stream based reports.
+         * For reports opened in write mode it finishes the reporting. For
+         * reports opened in read mode it disconnects from the source, any call
+         * waiting in getSpikes will be unblocked and getEndTime will return the
+         * last known timestamp.
+         *
+         * This function is thread-safe with getSpikes as it ensures that any
+         * thread waiting there with an infinite timeout will eventually return.
+         *
+         * Implicitly called by the destructor, (but calling the destructor and
+         * getSpikes at the same time is not safe!). Calling any other function
+         * after the report has been closed has undefined behavior.
+         */
+        void close();
```

Comment:
close() is intented to unlock any thread that may be potentially blocked with an infinite timeout in getSpikes(). Without this function the only alternative available for multithreaded programs that want to exit orderly is to always use a timeout in getSpikes, which is not very nice.